istioctl/cmd/istioctl/main.go

```diff
@@ -397,16 +416,16 @@ istioctl get virtualservice bookinfo
 			} else {
 				for _, typ := range typs {
 					typeConfigs, err := configClient.List(typ.Type, ns)
-					if err != nil {
-						return multierror.Prefix(err, fmt.Sprintf("Can't list %v:", typ.Type))
+					if err != nil && mustList[typ.Type] {
+						errs = multierror.Append(errs, multierror.Prefix(err, fmt.Sprintf("Can't list %v:", typ.Type)))
 					}
 					configs = append(configs, typeConfigs...)
```

Comment:
I think we should not `append` the error `typeConfigs`, otherwise, if all types for `typeConfigs` are in error state, then we will not get the message of `No resources found`

Reply:
We still get the message because line 427 prints `No resources found` before returning _err_.

Reply:
I think no because `if len(configs) == 0` will not work as `len(configs)` will `>0` if there are error typeconfigs.